pkg/pb/synthetic_monitoring/checks_extra.go

```diff
@@ -141,6 +157,10 @@ func (c *Check) Validate() error {
 	if c.Settings.Tcp != nil {
 		settingsCount++
 
+		if err := validateHost(c.Target); err != nil {
```

Comment:
for TCP, the target should be `host:port` , So we need to split into host and target first, then validate each.  ie port must be > 0 and <= 65535

Reply:
I completely missed that, thanks!

Fixed